nsqd/http.go

```diff
@@ -596,3 +600,41 @@ func (s *httpServer) printStats(stats []TopicStats, health string, startTime tim
 	}
 	return buf.Bytes()
 }
+
+type allowedOpts struct {
+	NSQLookupdTCPAddresses []string `json:"nsqlookupd_tcp_addresses"`
```

Comment:
Yes, I'm happy to discuss alternative APIs because now is the time to do it.  It's not that I disagree with the benefits, it's that I think we should try to come up with _an even better API_ if we want those semantics.

What about HTTP verbs?  What about individual endpoints?  If being more specific helps improve the API then I'm all for it...

Implementation wise, the HTTP side of this PR is the trivial stuff, most of the noise here is cleanup and proper handling of (now dynamic) state.